test/file_test.py

```diff
@@ -169,36 +169,64 @@ def test_move(self):
     def test_text(self):
         t = File(self.path, luigi.format.UTF8)
         a = u'我éçф'
-        with t.open('wb') as f:
+        with t.open('w') as f:
             f.write(a)
-        with t.open('rb') as f:
+        with t.open('r') as f:
             b = f.read()
         self.assertEqual(a, b)
 
     def test_format_chain(self):
-        UTF8WIN = luigi.format.TextWrapper(encoding='utf8', newline='\r\n')
+        UTF8WIN = luigi.format.TextFormat(encoding='utf8', newline='\r\n')
         t = File(self.path, UTF8WIN >> luigi.format.Gzip)
         a = u'我é\nçф'
 
-        with t.open('wb') as f:
+        with t.open('w') as f:
             f.write(a)
 
-        with gzip.open(self.path, 'rb') as f:
-            b = f.read()
+        f = gzip.open(self.path, 'r')
+        b = f.read()
+        f.close()
 
         self.assertEqual(b'\xe6\x88\x91\xc3\xa9\r\n\xc3\xa7\xd1\x84', b)
 
     def test_format_chain_reverse(self):
         t = File(self.path, luigi.format.UTF8 >> luigi.format.Gzip)
 
-        with gzip.open(self.path, 'wb') as f:
-            f.write(b'\xe6\x88\x91\xc3\xa9\r\n\xc3\xa7\xd1\x84')
+        f = gzip.open(self.path, 'w')
+        f.write(b'\xe6\x88\x91\xc3\xa9\r\n\xc3\xa7\xd1\x84')
+        f.close()
 
-        with t.open('rb') as f:
+        with t.open('r') as f:
             b = f.read()
 
         self.assertEqual(u'我é\nçф', b)
 
+    @mock.patch('os.linesep', '\r\n')
+    def test_format_newline(self):
+        t = File(self.path, luigi.format.SysNewLine)
+
+        with t.open('w') as f:
+            f.write(b'a\rb\nc\r\nd')
+
+        with t.open('r') as f:
+            b = f.read()
+
+        with open(self.path, 'rb') as f:
+            c = f.read()
+
+        self.assertEqual(b'a\nb\nc\nd', b)
+        self.assertEqual(b'a\r\nb\r\nc\r\nd', c)
+
+    def test_del_with_Text(self):
+        t = File(self.path, luigi.format.UTF8)
+        p = t.open('w')
+        print(u'test', file=p)
```

Comment:
No, it's that the io.TextIOWrapper try to close the file on `del` which we don't want since in our case close => commit